marge/app.py

```diff
@@ -27,138 +28,153 @@ def time_interval(s):
         raise argparse.ArgumentTypeError('Invalid time interval (e.g. 12[s|min|h]): %s', s)
 
 
-def _parse_args(args):
-    parser = argparse.ArgumentParser(description=__doc__)
-
-    def arg(name, env_var=True, **kwargs):
-        if env_var:
-            env_var_name = 'MARGE_' + name[2:].replace('-', '_').upper()
-            env_var_val = os.getenv(env_var_name)
-            if env_var_val:
-                if kwargs.get('action') in ['store_true', 'store_false']:
-                    env_var_val = env_var_val.lower() in ['true', 'yes', '1']
-                kwargs = dict(kwargs, default=env_var_val, required=False)
+def _parse_config(args):
 
-        parser.add_argument(name, **kwargs)
+    for bad_arg in ['--auth-token', '--ssh-key']:
+        if bad_arg in args:
+            raise Exception('"%s" can only be set via ENV var or config file.' % bad_arg)
 
     def regexp(s):
         try:
             return re.compile(s)
         except re.error as err:
             raise argparse.ArgumentTypeError('Invalid regexp: %r (%s)' % (s, err.msg))
 
-    arg(
+    parser = configargparse.ArgParser(
+        auto_env_var_prefix='MARGE_',
+        ignore_unknown_config_file_keys=True,  # Don't parse unknown args
+        config_file_parser_class=configargparse.YAMLConfigFileParser,
+        args_for_setting_config_path=["--config-file"],
+        formatter_class=configargparse.ArgumentDefaultsRawHelpFormatter,
+        description=__doc__,
+    )
+    auth_token_group = parser.add_mutually_exclusive_group(required=True)
+    auth_token_group.add_argument(
+        '--auth-token',
+        type=str,
+        metavar='TOKEN',
+        help=(
+            'Your gitlab token.\n'
+            'WARNING:\n'
+            'Due to security reasons, you can only set this via ENV var or config file.\n'
+        ),
+    )
+    auth_token_group.add_argument(
         '--auth-token-file',
         type=argparse.FileType('rt'),
         metavar='FILE',
-        help='Your gitlab token; must provide this flag or set MARGE_AUTH_TOKEN',
-        env_var=False,
+        help='Path to your gitlab token file.\n',
     )
-    arg(
+    parser.add_argument(
```

Comment:
I kinda prefer:
```
arg = parse.add_argument 
arg(...
```
but up to you.

Reply:
hmm, what's the reason?

Reply:
Matter of taste. Less indent, more concise, more readable IMO and makes tweaking easier. I'm fine w/ you leaving it as is though.